File: common/src/main/java/com/ssblur/scriptor/exceptions/InvalidGeneratorException.java
package com.ssblur.scriptor.exceptions;

import net.minecraft.resources.ResourceLocation;

public class InvalidGeneratorException extends RuntimeException {
  public InvalidGeneratorException(String generator, ResourceLocation location) {
    super(String.format("Generator '%s' does not exist or is not registered! (Used by %s)", generator, location));
  }
}

File: common/src/main/kotlin/com/ssblur/scriptor/error/InvalidGeneratorException.kt
package com.ssblur.scriptor.error

import net.minecraft.resources.ResourceLocation

class InvalidGeneratorException(generator: String?, location: ResourceLocation?) : RuntimeException(
    String.format(
        "Generator '%s' does not exist or is not registered! (Used by %s)",
        generator,
        location
    )
)
